explainability/explainability-core/src/main/java/org/kie/kogito/explainability/model/PerturbationContext.java

```diff
@@ -25,15 +26,25 @@
  */
 public class PerturbationContext {
 
+    private final Optional<Long> seed;
+
     private final Random random;
 
     private final int noOfPerturbations;
 
     public PerturbationContext(Random random, int noOfPerturbations) {
+        this.seed = Optional.empty();
         this.random = random;
         this.noOfPerturbations = noOfPerturbations;
     }
 
+    public PerturbationContext(Long seed, Random random, int noOfPerturbations) {
+        this.seed = Optional.of(seed);
```

Comment:
Just to be on the safe side
```suggestion
        this.seed = Optional.ofNullable(seed);
```